### ISSUE TITLE
"warn-as-error: true" in tspconfig.yaml does not work

### ISSUE THREAD

**Issue by mikeharder** [microsoft/typespec#2190]
`warn-as-error: true` in tspconfig.yaml doesn't appear to be working:

```
$ git diff
--- a/specification/contosowidgetmanager/Contoso.WidgetManager/main.tsp
-@doc("Versions info.")
+// @doc("Versions info.")

--- a/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml
-parameters:
+warn-as-error: true
+parameters:

$ npx tsp compile .
warning @azure-tools/typespec-azure-core/documentation-required

$ npx tsp compile . --warn-as-error
error @azure-tools/typespec-azure-core/documentation-required
```

**Comment by mikeharder** [comment]
Verified fixed in `@typespec/compiler@0.47.0-dev.8`.

**Comment by markcowl** [comment]
## Breaking Change Review
**Describe the breaking change**
 The emitter configuration for `warn-as-error`, which treated warnings as errors during compilation was silently ignored.  This change enables `warn-as-error` configuration in `tspconfig.yaml`.  Any specs that had this configuration and existing warnings would now return an error exit code if any warnings were emitted during compilation.

```yaml
- warn-as-error: true
```

**Back-compat design to avoid immediate breaking**
Any back-compat design: None

**Impact**
This change affecting:

- [ ] ARM
- [ ] Data-plane
- [x] Existing specs: 0 (checked in)
- [ ] New specs
- [ ] Emitters
- [ ] SDK

**Targeted Sprint**
Sprint: August (release: 8/8)

**Additional information**
Add any other information about the change including fixes for here.
